### DISCUSSION TITLE
Possible to use with Seeeduino XIAO ESP32S3 and an I2C display?

### DISCUSSION THREAD

**Discussion by rbuckton** [moononournation/Arduino_GFX#323] [Q&A]
I'm currently using Adafruit_GFX and https://github.com/limengdu/Arduino_ST7789_Fast/tree/master, but was thinking about using this library instead. The display I'm using is a [Seeedstudio Grove 1.2" IPS Display](https://us.seeedstudio.com/Grove-1-2-Inch-IPS-Display-p-5699.html), which is an ST7789 OLED display and employs an I2C connection rather than SPI. There doesn't appear to be an existing `Arduino_DataBus` subclass that will work. Would you consider adding a DataBus subclass to support this display, or do I need to try to write my own?

**Comment by moononournation** [comment]
The display in your link should be 9-bit SPI but not I2C. I have tried similar display before and only can initial success randomly. The SPI display without CS, DC and RST at the same time is not easy to ensure the initialization. My final solution is breakout the CS pin to ensure it works.

**Comment by rbuckton** [comment]
The display is marketed as an I2C device and seems to operate only over SDA/SCL. There's no access to the CS pin on the board, at least not without probably removing the display. I've almost got it working with a custom Data Bus and a subclass of `Android_ST7789` with slightly different init commands based on the ones in `Arduino_ST7789_Fast`. Clearing the screen and drawing shapes works, but writing text does not. When the characters are drawn to the screen they are partially drawn with small white vertical lines cutting through the characters. I'm guessing it may be due to cutting out some of the init commands, but I haven't had time to investigate fully yet.

The custom Data Bus I put together is very similar to `Arduino_XL9535SWSPI`, except it talks directly to SDA/SCL rather than using `Wire` to find `XL9535`, with SDA in place of MOSI and SCL in place of SCK.

I will admit I'm fairly new to Arduino so I'm not familiar with the ins and outs of some of the protocols like I2C and SPI, so I'm not sure what 9-bit SPI is and how it relates to this board's operation.
